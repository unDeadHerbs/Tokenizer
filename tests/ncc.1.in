the cat + 1 2 45 ~= 2.3 1e3 2e-12 .1e3;
<<- a block comment
    as a note: our parser won't be able to deal with long tokens
    if there are invalid partial tokens we can't backup more than 1 and so must error
    example: 1e+f as the e and the + are two steps into a real's parse
    example: <<j as the << could be a <<- we have to have consumed two chars after the first and we can't backup
    example: "\ug is a unicode string with invalid argument, the \ug requires consuming three tokens
->>
"another token"
"string with unicode\u21 \u222Be\u207Fdn"
error "\uc\ug\uc"
# line comment
handeled 1e+f
handeled <<j
handeled <<<- comment with less before it ->>
# line comment withouth newline
