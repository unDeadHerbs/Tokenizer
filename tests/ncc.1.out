| Token Type | Token Value
+------------+-- - - -  -  -
|      IDENT | 'the' at 1:1
|      IDENT | 'cat' at 1:6
|       PLUS | '+' at 1:11
|        INT | '1' at 1:13
|        INT | '2' at 1:16
|        INT | '45' at 1:19
|  NOT_EQUAL | '~=' at 1:23
|       REAL | '2.3 ' at 1:26
|       REAL | '1e3' at 1:30
|       REAL | '2e-12' at 1:35
|       REAL | '.1e3' at 1:42
|  SEMICOLON | ';' at 1:48
|     STRING | 'another token' at 9:1
|     STRING | 'string with unicode! ∫eⁿdn' at 10:1
|      IDENT | 'error' at 11:1
|     STRING | 'g' at 11:8
|      ERROR | 'bad Unicode' at 11:16
|      IDENT | 'handeled' at 13:1
|        INT | '1' at 13:11
|      IDENT | 'e' at 13:15
|       PLUS | '+' at 13:17
|      IDENT | 'f' at 13:18
|      IDENT | 'handeled' at 15:1
|       LESS | '<' at 15:11
|       LESS | '<' at 15:14
|      IDENT | 'j' at 15:16
|      IDENT | 'handeled' at 17:1
|       LESS | '<' at 17:11
+------------+-- - - -  -  -
